### PR TITLE
Testing invocation of math functions inside target region.

### PR DESCRIPTION
**Pull request by spophale**
QMCPACK team suggested this test as it is important for them that math functions are supported in offloaded kernels.

### CODE REVIEW

File: tests/4.5/application_kernels/qmcpack_target_math.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
+//===--- qmcpack_target_math.c --- math lib invocation inside target---------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+//
```
Comment (josemonsalve2) [inline tests/4.5/application_kernels/qmcpack_target_math.c]: Add description

**Comment by josemonsalve2** [comment]
I don't think we should have math.h tests to be part of the specification tests. We cannot claim that an implementation is not compliant with the specification if it does not run math.h functions. 

At some point we discussed separating this sort of tests that are not "specification" tests but "application specific tests".

File: tests/4.5/application_kernels/qmcpack_target_math.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  OMPVV_INFOMSG("test_math_lib_inside_target");
+
+  double array[N];
+  int errors = 0;
+  
+  // Array initialization
+  for (int i = 0; i < N; ++i) {
+    array[i] = 0.99;
+  }
+
```
Comment (ye-luo) [inline tests/4.5/application_kernels/qmcpack_target_math.c]: Let us make this test closer to the real code and more challenging.
Please a dummy line
`int c99_zero = FP_ZERO;`
This should choke clang in the CUDA pass.

Comment (spophale) [reply tests/4.5/application_kernels/qmcpack_target_math.c]: FP_ZERO is for floating point. Is there some reason you want to assign it to int ?

Comment (ye-luo) [reply tests/4.5/application_kernels/qmcpack_target_math.c]: FP_ZERO is one of the return value of fpclassify, the return type is int.
https://www.gnu.org/software/libc/manual/html_node/Floating-Point-Classes.html

**Comment by tmh97** [comment]
@spophale is this PR ready for merge?

**Comment by spophale** [comment]
I want to separate the two tests. Then merge.

**Comment by ye-luo** [comment]
> I want to separate the two tests. Then merge.

Maybe it is better to merge this one in a timely fashion and then make a new PR to split the test.

**Pull request merged**